test/Smart-Account-Factory.specs.ts

```diff
@@ -20,40 +21,68 @@ describe("Smart Account Factory", async () => {
       smartAccountFactory: await getSmartAccountFactory(),
       mockToken: mockToken,
       ecdsaModule: ecdsaModule,
+      entryPoint: entryPoint,
     };
   });
   describe("Constructor", async () => {
 
     it ("reverts with zero address provided as implementation", async () => {
       const SmartAccountFactory = await ethers.getContractFactory("SmartAccountFactory");
       await expect(
-        SmartAccountFactory.deploy(AddressZero)
+        SmartAccountFactory.deploy(AddressZero, deployer.address)
       ).to.be.revertedWith("implementation cannot be zero");
     });
 
     it ("sets non-zero address implementation", async () => {
       const { smartAccountImplementation } = await setupTests();
       const SmartAccountFactory = await ethers.getContractFactory("SmartAccountFactory");
-      const testFactory = await SmartAccountFactory.deploy(smartAccountImplementation.address);
+      const testFactory = await SmartAccountFactory.deploy(smartAccountImplementation.address, deployer.address);
       await testFactory.deployed();
       expect(await testFactory.basicImplementation()).to.equal(smartAccountImplementation.address);
     });
 
     it ("deploys Default Callback Handler instance", async () => {
       const { smartAccountImplementation } = await setupTests();
       const SmartAccountFactory = await ethers.getContractFactory("SmartAccountFactory");
-      const testFactory = await SmartAccountFactory.deploy(smartAccountImplementation.address);
+      const testFactory = await SmartAccountFactory.deploy(smartAccountImplementation.address, deployer.address);
       await testFactory.deployed();
       const callbackHandlerAddress = await testFactory.minimalHandler();
       const callbackHandler = await ethers.getContractAt("DefaultCallbackHandler", callbackHandlerAddress);
       expect(await callbackHandler.NAME()).to.equal("Default Callback Handler");
     });
 
+    it ("successfully changes owner to a new one", async () => {
+      const { smartAccountImplementation } = await setupTests();
+      const SmartAccountFactory = await ethers.getContractFactory("SmartAccountFactory");
+      const testFactory = await SmartAccountFactory.deploy(smartAccountImplementation.address, charlie.address);
+      await testFactory.deployed();
+      expect(await testFactory.owner()).to.equal(charlie.address);
+      expect(await testFactory.owner()).to.not.equal(deployer.address);
+    });
+  });
+
+  describe("Stakeable", async () => {
+    it ("can add stake to the EP", async () => {
+      const { smartAccountFactory, smartAccountImplementation, entryPoint } = await setupTests();
+      const stakeAmount = ethers.utils.parseEther("1.234256");
+      await smartAccountFactory.addStake(entryPoint.address, 600, { value: stakeAmount });
+      const depositInfo = await entryPoint.getDepositInfo(smartAccountFactory.address);
+      expect(depositInfo.stake).to.equal(stakeAmount);
+    });
+
+    // can unlock
```

Comment:
Unimplemented test cases. Should implement these test cases as well

Reply:
Yes, I will pass them to Darpit.

Reply:
I suggest merging in without those tests, as those methods just forward everything to the EP. 
Until it is merged, I can not stake the factory to test Multichain Validator via bundler properly.